pkg/plugins/manager/client/client.go

```diff
@@ -204,3 +221,21 @@ func (s *Service) plugin(ctx context.Context, pluginID string) (*plugins.Plugin,
 
 	return p, true
 }
+
+func (s *Service) attachJWT(ctx context.Context, pluginCtx backend.PluginContext) context.Context {
```

Comment:
This feels like a good candidate for a new plugin client middleware https://github.com/grafana/grafana/blob/6dbe3b555f5cc6de2e3ada486f89cf9c5ef31336/pkg/plugins/ifaces.go#L93.

You can find a good example [here](https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go#L40)